packages/reflect-yjs/src/mutators.ts

```diff
@@ -20,24 +20,92 @@ export async function updateYJS(
   tx: WriteTransaction,
   {name, update}: {name: string; update: string},
 ) {
-  const existing = await tx.get<string>(yjsProviderServerKey(name));
-  const getKeyToSet =
-    tx.location === 'client' ? yjsProviderClientKey : yjsProviderServerKey;
+  const existing = await getServerUpdate(name, tx);
+  const set = tx.location === 'client' ? setClientUpdate : setServerUpdate;
   if (!existing) {
-    await tx.set(getKeyToSet(name), update);
+    await set(name, update, tx);
   } else {
     const updates = [base64.toByteArray(existing), base64.toByteArray(update)];
     const merged = Y.mergeUpdatesV2(updates);
-    await tx.set(getKeyToSet(name), base64.fromByteArray(merged));
+    await set(name, base64.fromByteArray(merged), tx);
   }
 }
 
-export function yjsProviderClientKey(name: string): string {
+function yjsProviderClientUpdateKey(name: string): string {
   return `yjs/provider/client/${name}`;
 }
 
-export function yjsProviderServerKey(name: string): string {
-  return `yjs/provider/server/${name}`;
+function yjsProviderServerUpdatePrefix(name: string): string {
+  return `yjs/provider/server/${name}/`;
+}
+
+function setClientUpdate(name: string, update: string, tx: WriteTransaction) {
+  return tx.set(yjsProviderClientUpdateKey(name), update);
+}
+
+async function setServerUpdate(
+  name: string,
+  update: string,
+  tx: WriteTransaction,
+) {
+  const writes = [];
+  let i = 0;
+  for (; i * CHUNK_LENGTH < update.length; i++) {
+    const existing = await tx.get(yjsProviderServerKey(name, i));
+    const chunk = update.substring(
+      i * CHUNK_LENGTH,
+      i * CHUNK_LENGTH + CHUNK_LENGTH,
+    );
+    if (existing !== chunk) {
+      writes.push(
+        tx.set(
+          yjsProviderServerKey(name, i),
+          update.substring(i * CHUNK_LENGTH, i * CHUNK_LENGTH + CHUNK_LENGTH),
+        ),
+      );
+    }
+  }
+  for await (const key of tx
```

Comment:
This took me some time to understand.

Please add a comment that we are deleting the rest in case the old value was larger.

Reply:
done